Lib/idlelib/codecontext.py

```diff
@@ -22,15 +22,37 @@
 UPDATEINTERVAL = 100 # millisec
 FONTUPDATEINTERVAL = 1000 # millisec
 
+
 def getspacesfirstword(s, c=re.compile(r"^(\s*)(\w*)")):
+    "Extract the beginning whitespace and first word from s."
     return c.match(s).groups()
 
 
 class CodeContext:
+    "Display block context above the edit window."
+
     bgcolor = "LightGray"
     fgcolor = "Black"
 
     def __init__(self, editwin):
+        """Initialize settings for context block.
+
+        Args:
+            editwin: Editor window for the context block.
+
+        self.text is the editor window text widget.
+
+        self.textfont is the editor window font.
+
+        self.label displays the code context text at the top of the
+        editor.  It is initialized to None to not display the code context
+        and it is toggled via <<toggle-code-context>>.  If code context is
+        not suitable for an editor window, then unbind the text from
+        <<toggle-code-context>>.
```

Comment:
> about whether binding for some features should be in the feature init.

I tried this and it worked, but I think it should be its own issue since it's more than just code cleanup.